src/common/libflux/event.h

```diff
@@ -1,22 +1,30 @@
 #ifndef _FLUX_CORE_EVENT_H
 #define _FLUX_CORE_EVENT_H
 
-#include <json.h>
 #include <stdbool.h>
+#include <stdarg.h>
 #include <czmq.h>
 
 #include "handle.h"
 
-/* Decode event message, setting 'in' to JSON payload.
- * If payload is missing, fail with errno == EPROTO.
+/* Decode an event message.
+ * If topic is non-NULL, assign the event topic string.
+ * If json_str is non-NULL, assign the payload.  json_str indicates whether
+ * payload is expected and it is an EPROTO error if they don't match.
  * Returns 0 on success, or -1 on failure with errno set.
  */
-int flux_json_event_decode (zmsg_t *zmsg, json_object **in);
+int flux_event_decode (zmsg_t *zmsg, const char **topic, const char **json_str);
+
+/* Encode an event message.
+ * If json_str is non-NULL, it is copied to the message payload.
+ * Returns message or NULL on failure with errno set.
+ */
+zmsg_t *flux_event_encode (const char *topic, const char *json_str);
 
 /* Send/receive events
  */
-int flux_event_send (flux_t h, json_object *request, const char *fmt, ...);
-int flux_event_recv (flux_t h, json_object **respp, char **topic, bool nb);
+int flux_event_send (flux_t h, zmsg_t **zmsg);
```

Comment:
The **zmsg makes me curious, is that a side-effect of zmq de-allocating the message on send?